packages/react-aria-components/src/Menu.tsx

```diff
@@ -268,3 +328,86 @@ function MenuItemInner<T>({item}: MenuItemInnerProps<T>) {
     </ElementType>
   );
 }
+
+interface MenuItemTriggerInnerProps<T> {
+  item: Node<T>,
+  popover: ReactElement,
+  parentMenuRef: RefObject<HTMLDivElement>
+}
+
+function MenuItemTriggerInner<T>({item, popover, parentMenuRef}: MenuItemTriggerInnerProps<T>) {
+  let state = useContext(MenuStateContext)!;
+  let ref = useObjectRef<any>(item.props.ref);
+  let rootMenuTriggerState = useContext(RootMenuTriggerStateContext)!;
+  let submenuContext = useContext(SubmenuContext)!;
+  let submenuTriggerState = UNSTABLE_useSubmenuTriggerState({triggerKey: item.key}, rootMenuTriggerState);
+  let submenuRef = useRef<HTMLDivElement>(null);
+  let {submenuTriggerProps, submenuProps, popoverProps} = UNSTABLE_useSubmenuTrigger({
+    node: item,
+    parentMenuRef,
+    submenuRef
+  }, submenuTriggerState, ref);
+  let {menuItemProps, labelProps, descriptionProps, keyboardShortcutProps, ...states} = useMenuItem({
+    key: item.key,
+    ...submenuTriggerProps
+  }, state, ref);
+  let props: MenuItemProps<T> = item.props;
+  let {isFocusVisible, focusProps} = useFocusRing();
+  let isSubmenuOpen = submenuTriggerState.isOpen;
+  let renderProps = useRenderProps({
+    ...props,
+    id: undefined,
+    children: item.rendered,
+    defaultClassName: 'react-aria-MenuItem',
+    values: {
+      ...states,
+      isHovered: states.isFocused,
```

Comment:
I forget, do we want isHovered to be true on keyboard focus? I have a faint recollection that we changed this before the break for some other components but that there was some complications. I see that we have the same for MenuItem as well

Reply:
I feel like we changed this somewhere else. I'll try to find where this was discussed.

Reply:
Looks like we made a change in listbox regarding this: https://github.com/adobe/react-spectrum/commit/2b05b6d984e1b37d4ac76f0bb305433d8f57d988#diff-9da69535256428710b8d6d7b6ce51163e15105e3084c7bc06d2ba612e1ba9dbf

I'll update these to use the hover state from useHover instead.